test/Dialect/SystemC/canonicalize.mlir

```diff
@@ -0,0 +1,133 @@
+
+  %16 = systemc.convert %uint32 : (i32) -> !systemc.unsigned
+  %c16 = systemc.convert %16 : (!systemc.unsigned) -> i32
+
+  %17 = systemc.convert %uint32 : (i32) -> !systemc.uint_base
+  %c17 = systemc.convert %17 : (!systemc.uint_base) -> i32
+
+  // We don't necessarily know the max bit-width of unsigned, thus we should not fold above 64 bits
+  // CHECK-NEXT: [[V12:%.+]] = systemc.convert %uint65 : (i65) -> !systemc.unsigned
+  // CHECK-NEXT: [[V13:%.+]] = systemc.convert [[V12]] : (!systemc.unsigned) -> i65
+  %18 = systemc.convert %uint65 : (i65) -> !systemc.unsigned
+  %c18 = systemc.convert %18 : (!systemc.unsigned) -> i65
+
+  // CHECK-NEXT: systemc.cpp.assign %scuint32 = %scuint32 : !systemc.uint<32>
+  // CHECK-NEXT: systemc.cpp.assign %scuint32 = %scuint32 : !systemc.uint<32>
+  // CHECK-NEXT: systemc.cpp.assign %uint32 = %uint32 : i32
+  // CHECK-NEXT: systemc.cpp.assign %uint16 = [[V1]] : i16
+  // CHECK-NEXT: systemc.cpp.assign %scint32 = %scint32 : !systemc.int<32>
+  // CHECK-NEXT: systemc.cpp.assign %scint32 = [[V3]] : !systemc.int<32>
+  // CHECK-NEXT: systemc.cpp.assign %scuint32 = [[V5]] : !systemc.uint<32>
+  // CHECK-NEXT: systemc.cpp.assign %logic = [[V7]] : !systemc.logic
+  // CHECK-NEXT: systemc.cpp.assign %bool = %bool : i1
+  // CHECK-NEXT: systemc.cpp.assign %uint32 = [[V9]] : i32
+  // CHECK-NEXT: systemc.cpp.assign %logic = %logic : !systemc.logic
+  // CHECK-NEXT: systemc.cpp.assign %uint_base = %uint_base : !systemc.uint_base
+  // CHECK-NEXT: systemc.cpp.assign %uint_base = [[V11]] : !systemc.uint_base
+  // CHECK-NEXT: systemc.cpp.assign %uint_base = %uint_base : !systemc.uint_base
+  // CHECK-NEXT: systemc.cpp.assign %uint32 = %uint32 : i32
+  // CHECK-NEXT: systemc.cpp.assign %uint_base = %uint_base : !systemc.uint_base
+  // CHECK-NEXT: systemc.cpp.assign %uint32 = %uint32 : i32
+  // CHECK-NEXT: systemc.cpp.assign %uint32 = %uint32 : i32
+  // CHECK-NEXT: systemc.cpp.assign %uint65 = [[V13]] : i65
+  systemc.cpp.assign %scuint32 = %0 : !systemc.uint<32>
+  systemc.cpp.assign %scuint32 = %c1 : !systemc.uint<32>
+  systemc.cpp.assign %uint32 = %c2 : i32
+  systemc.cpp.assign %uint16 = %c3 : i16
+  systemc.cpp.assign %scint32 = %c4 : !systemc.int<32>
+  systemc.cpp.assign %scint32 = %c5 : !systemc.int<32>
+  systemc.cpp.assign %scuint32 = %c6 : !systemc.uint<32>
+  systemc.cpp.assign %logic = %c7 : !systemc.logic
+  systemc.cpp.assign %bool = %c8 : i1
+  systemc.cpp.assign %uint32 = %c9 : i32
+  systemc.cpp.assign %logic = %c10 : !systemc.logic
+  systemc.cpp.assign %uint_base = %c11 : !systemc.uint_base
+  systemc.cpp.assign %uint_base = %c12 : !systemc.uint_base
+  systemc.cpp.assign %uint_base = %c13 : !systemc.uint_base
+  systemc.cpp.assign %uint32 = %c14 : i32
+  systemc.cpp.assign %uint_base = %c15 : !systemc.uint_base
+  systemc.cpp.assign %uint32 = %c16 : i32
+  systemc.cpp.assign %uint32 = %c17 : i32
+  systemc.cpp.assign %uint65 = %c18 : i65
```

Comment:
This test covers a ton! It may be more readable by either using more descriptive names, e.g., `%c8` could be `%convert_logic_to_i1` and/or by moving SSA values closer to their use sites.